misc/python/materialize/mzcompose/composition.py

```diff
@@ -623,15 +623,15 @@ def exec(
         )
 
     def pull_if_variable(self, services: list[str], max_tries: int = 2) -> None:
-        """Pull fresh service images in case the tag indicates thee underlying image may change over time.
+        """Pull fresh service images in case the tag indicates the underlying image may change over time.
 
         Args:
             services: List of service names
         """
 
         for service in services:
             if "image" in self.compose["services"][service] and any(
-                self.compose["services"][service]["image"].endswith(tag)
+                self.compose["services"][service]["image"].contains(tag)
```

Comment:
Otherwise, something like `:unstable-1f03917846a81a8f548447e90231374c04a8657d` won't be matched.